[WebGL2] Don't apply buffer offset twice in getBufferSubData

djg opened the pull request:
#### 747aaf24c4e99c2fcb28a00c8620072a834874a0
<pre>
[WebGL2] Don&apos;t apply buffer offset twice in getBufferSubData
<a href="https://bugs.webkit.org/show_bug.cgi?id=243566">https://bugs.webkit.org/show_bug.cgi?id=243566</a>
rdar://98154048

Reviewed by Kimmo Kinnunen.

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::getBufferSubData):
baseAddress() already takes care of offset into ArrayBuffer.

Canonical link: <a href="https://commits.webkit.org/253175@main">https://commits.webkit.org/253175@main</a>
</pre>

kkinnunen-apple commented on LayoutTests/fast/canvas/webgl/webgl2-getbuffersubdata.html:
```diff
@@ -0,0 +1,80 @@
+description("Make sure that reading from WebGL2 buffers into offset ArrayBufferView works as expected.");
+
+if (window.internals)
+	internals.settings.setWebGL2Enabled(true);
```
I think you could use 2 or 4 space indent, adding yet another indent convention may be detrimental.

djg replied:
Converted tabs to 4 spaces.

djg committed:
[WebGL2] Don't apply buffer offset twice in getBufferSubData
https://bugs.webkit.org/show_bug.cgi?id=243566
rdar://98154048

Reviewed by Kimmo Kinnunen.

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::getBufferSubData):
baseAddress() already takes care of offset into ArrayBuffer.

Canonical link: https://commits.webkit.org/253175@main
